src/Docker/Client/Types.hs

```diff
@@ -1391,7 +1392,9 @@ instance FromJSON ContainerConfig where
         image <- o .: "Image"
         volumes <- o .: "Volumes"
         workingDir <- o .:? "WorkingDir"
-        entrypoint <- o .:? "Entrypoint"
+        entrypoint <- asum [ o .:? "Entrypoint" .!= []
+                           , (:[]) <$> o .: "Entrypoint"
```

Comment:
Thanks for your comment!

`(:[]) <$> o .:? "Entrypoint" .!= []
`
The code above does not work as it gives rise to `[[Text]]` instead of `[Text]`.

I tested my patch manually and confirmed that it works correctly for all three types. I will create a new PR including test cases!